unlock-app/src/components/creator/members/GrantKeysDrawer.tsx

```diff
@@ -0,0 +1,274 @@
+import React, { useEffect, useState, useContext } from 'react'
+import { useForm } from 'react-hook-form'
+import { Locks, Lock } from '../../../unlockTypes'
+import Drawer from '../../interface/Drawer'
+import { WalletServiceContext } from '../../../utils/withWalletService'
+import { Web3ServiceContext } from '../../../utils/withWeb3Service'
+import { AuthenticationContext } from '../../../contexts/AuthenticationContext'
+import { ConfigContext } from '../../../utils/withConfig'
+
+import {
+  Input,
+  Label,
+  Select,
+  LoadingButton,
+  Button,
+} from '../../interface/checkout/FormStyles'
+import { ACCOUNT_REGEXP } from '../../../constants'
+
+interface GrantKeyFormProps {
+  lock: Lock
+  onGranted: (granted: boolean) => void
+}
+
+// Prevents re-rendering when time changes!
+const now = new Date().getTime()
+
+/**
+ * https://stackoverflow.com/questions/30166338/setting-value-of-datetime-local-from-date
+ * @param date
+ * @returns
+ */
+const formatDate = (date: Date) => {
+  date.setMinutes(date.getMinutes() - date.getTimezoneOffset())
+  return date.toISOString().slice(0, 16)
+}
+
+/**
+ * Form part
+ * @returns
+ */
+const GrantKeyForm = ({ onGranted, lock }: GrantKeyFormProps) => {
+  const { account, network } = useContext(AuthenticationContext)
+  const config: any = useContext(ConfigContext)
+
+  const walletService = useContext(WalletServiceContext)
+  const web3Service = useContext(Web3ServiceContext)
+
+  const [transaction, setTransaction] = useState<string>('')
+  const [loading, setLoading] = useState(false)
+
+  const defaultValues = {
+    recipient: '',
+    expiration: formatDate(new Date(now + lock.expirationDuration * 1000)),
+    keyManager: '',
+  }
+
+  const {
+    handleSubmit,
+    register,
+    reset,
+    formState: { errors },
+  } = useForm({
+    defaultValues,
+  })
+
+  useEffect(() => reset(defaultValues), [lock.name])
+
+  interface onSubmitInterface {
+    recipient: string
+    keyManager: string
+    expiration: string
+  }
+
+  const onSubmit = async ({
+    recipient,
+    keyManager,
+    expiration,
+  }: onSubmitInterface) => {
+    setLoading(true)
+    try {
+      const existingExpiration =
+        await web3Service.getKeyExpirationByLockForOwner(
+          lock.address,
+          recipient,
+          network
+        )
+      if (existingExpiration > new Date().getTime() / 1000) {
+        alert(
+          'This address already owns a valid key. You cannot grant them a new one.'
+        )
```

Comment:
You know what, that's a great idea. I am going to add am `alert` component that we can use instead of the native ones. I'll open another PR and use it there.

Reply:
Built and replaced!